CONTRIBUTING.md

```diff
@@ -29,26 +31,119 @@ The [PyScript Code of Conduct](https://github.com/pyscript/governance/blob/main/
+
+To contribute code changes or bug fixes to PyScript, follow these steps to open a pull request:
+
+1. Fork the PyScript repository to your GitHub account.
+2. Create a new branch from the `main` branch with a descriptive name for your changes.
+3. Make your desired changes, ensuring they adhere to the project's coding styles (detailed in this document).
+4. Write tests for your changes to maintain code quality and avoid regressions.
+5. Commit your changes with clear and concise commit messages.
+6. Push your branch to your forked repository.
+7. Open a pull request (PR) from your branch to the `main` branch of the PyScript repository.
+8. Provide a clear and descriptive title for your PR.
+9. Include a detailed description of the changes made, along with any relevant context or references.
+10. Wait for feedback and engage in discussions to address any reviewer comments or change requests.
+
+**Important TIP**: All PRs should have a related issue. This is particularly important if the PR adds new functionality or changes some behavior. It can be very frustrating to work on something and then learn that not everyone shares the same opinion on how that specific problem should be solved. Creating an issue/discussion and discussing the problem before making a PR is the best way to avoid wasted effort.
+
+### PR approval requirements
+
+After you've created a PR there are a few extra steps that need to be taken so the PR can be approved. These steps aim to make sure the approval process is clear and that when a PR is merged, the level of quality of the codebase meets the standards set by the project `maintainers` in order to avoid regression and, ultimately, to keep high standards for the project users.
+
+If your PR changes or adds functionality, it's expected that it includes:
+
+**Description**: Every PR should have a meaningful and exhaustive description of what they do and a link to the original issue they address.
+
+**Title**: A descriptive title for your PR will help others find and refer to it consistently.
+
+**Tests**: This is the way we ensure added/modified functionality meets the related feature expectations and that other code changes don't break functionlity over time.
+
+**Documentation**: If your PR changes or adds functionality, it's expected that you also make sure to add the right information to the docs. For more information about how to work on the project documentation check out the [Setting Up Your Documentation Environment](https://github.com/pyscript/pyscript/blob/main/docs/development/setting-up-environment.md#setting-up-your-documentation-environment).
+
+Additionally, PR's which change user-facing functionality are expected to include an entry in the Changelog, for users' quick reference when comparing versions of PyScript. For information on updating the changelog, see [Updating the Changelog](https://github.com/pyscript/pyscript/blob/main/docs/development/developing.md#updating-the-changelog).
```

Comment:
OT: same as previous content ... this it *not* now changelog or docs changes usually land on a project.